Metallum/Classes/Controllers/NetworkManager.swift

```diff
@@ -0,0 +1,114 @@
+//
+//  NetworkManager.swift
+//  Metallum
+//
+//  Created by AppleTree on 3/3/16.
+//  Copyright © 2016 OlhaF. All rights reserved.
+//
+
+import Foundation
+
+final class NetworkManager {
+    
+    static let sharedInstance = NetworkManager()
+    
+    let session: NSURLSession
+    var jsonDictionary: NSDictionary?
+    
+    private init() {
+        let config = NSURLSessionConfiguration.defaultSessionConfiguration()
+        config.timeoutIntervalForRequest = 30.0
+        config.timeoutIntervalForResource = 60.0
+        config.HTTPAdditionalHeaders = ["application-id" : "D65ECD64-92B7-4FC8-FF53-CC8CB9314200",
+                                        "secret-key" : "663FD35E-769A-3B6A-FF9D-7C3376D3C200",
+                                        "application-type" : "REST"]
+        
+        session = NSURLSession(configuration: config)
+    }
+    
+    func downloadData(complerionHandler completion: ()->Void) {
+        let resourceURL = NSURL(string: "https://api.backendless.com/v1/data/Band")
+        let currentRequest = NSURLRequest(URL: resourceURL!)
+        
+        let task = session.dataTaskWithRequest(currentRequest) { (let data, let response, let error) -> Void in
+            if ((response?.respondsToSelector("statusCode")) != nil) {
+                if let httpResponse = response as? NSHTTPURLResponse {
+                    if httpResponse.statusCode == 200 {
+                        self.parseJSONFromData(data!)
+                        if self.jsonDictionary != nil {
```

Comment:
Do not store JSON in NetworkManager. It is not its job. Pass it into completion handler